libraries/AC_Planck/AC_Planck.cpp

```diff
@@ -286,3 +339,86 @@ bool AC_Planck::get_posvel_cmd(Location &loc, Vector3f &vel_cms, float &yaw_cd,
   _cmd.is_new = false;
   return true;
 }
+
+bool AC_Planck::check_for_high_tension_timeout() {
+  //No failure if not flying
+  if(AP_Motors::get_singleton()->get_spool_state() == AP_Motors::SpoolState::SHUT_DOWN) {
+    return false;
+  }
+
+  //No comms from the tether
+  bool tether_comms_failed = is_tether_timed_out();
+
+  //Determine if high tension has been or should have been triggered
+  bool high_tension_triggered = _tether_status.high_tension || tether_comms_failed;
+
+  //Hasn't failed if not triggered
+  if(!high_tension_triggered) {
+    _tether_status.sent_failed_message = false;
+    return false;
+  }
+
+  //The amount of time to wait for high tension to timeout is a function of
+  //initial altitude when the high tension event ocurred. Use tag altitude if available
+  float timeout_s = 0;
+  const float reel_rate_cms = 38.1; //~1.25ft/s
+  if(!is_equal(_tether_status.high_tension_tag_alt_cm,0.0f)) {
+    timeout_s = _tether_status.high_tension_tag_alt_cm / reel_rate_cms;
+  } else {
+    timeout_s = _tether_status.high_tension_alt_cm / reel_rate_cms;
+  }
+
+  //Account for the 10s spent in "locked" mode, but only in a comms or position loss state
+  bool pos_reference_good = get_commbox_state() || get_tag_tracking_state();
```

Comment:
Maybe add a comment saying we only check commbox since that's what the DECK uses, and possibly will change in the future

Reply:
Done.